conans/test/unittests/model/version/test_version_range_intersection.py

```diff
@@ -7,34 +7,40 @@
     ['>1.0', ">1.0", ">1.0"],
     ['>=1.0', ">1.0", ">1.0"],
     ['>1.0', ">1.1", ">1.1"],
-    ['>1.0', ">=1.1", ">=1.1"],
-    ['>=1.0', ">=1.1", ">=1.1"],
+    ['>1.0', ">=1.1", ">=1.1-"],
```

Comment:
I am not sure why the result of the intersection has changed.

Reply:
Because after the changes in #14814, the canonical representation of >=1.1 is >=1.1-, which is what the conditions now store and so what gets used in the calculations of the intersection

Reply:
(Note that I updated the branch be in sync with release/2.0)

Reply:
Ok, on one hand it makes sense, it is expected, on the other hand it made me a bit uncomfortable for some reason, doesn't read very intuitive that the intersection apparently changes the limits, it should be kind of orthogonal. I have had a look, it is possible to change it, by storing the original version in the ``Condition`` object, but probably it is not worth it, so lets leave it as-is.

Reply:
I think it might be worth pursuing the change? I share the feeling that it might be counterintuitive for users to see that the ranges are changing. Storing the original version expression seems doable to me, and tests seem to be passing locally. Will discuss later